app/src/main/java/com/teamagam/gimelgimel/app/model/ViewsModels/Message.java

```diff
@@ -1,80 +1,69 @@
 package com.teamagam.gimelgimel.app.model.ViewsModels;
 
+import android.support.annotation.StringDef;
+
 import com.google.gson.annotations.SerializedName;
 
 import java.util.Date;
 
-/**
- * Created by Gil.Raytan on 21-Mar-16.
- */
 
 /**
  * A class representing a type of ic_message passed to the server
  */
-public class Message<T extends MessageContentInterface>{
+public abstract class Message<T>{
+
+    @StringDef({TEXT, LAT_LONG, USER_LOCATION})
```

Comment:
You should add the @Retention annotation to explicitly define the retention policy.

IMO its a SOURCE retention policy (at compile time - to avoid performance issues)